shared/reducers/fs.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import * as FSGen from '../actions/fs-gen'
+import * as Constants from '../constants/fs'
+import * as Types from '../constants/types/fs'
+
+const initialState = Constants.makeState()
+
+export default function(state: Types.State = initialState, action: FSGen.Actions) {
+  switch (action.type) {
+    case FSGen.resetStore:
+      return initialState
+    case FSGen.increaseCount: {
+      const {amount = 1} = action.payload
```

Comment:
What's going on here? Having trouble detangling this syntax.

Reply:
sure. this is destructuring (https://babeljs.io/learn-es2015/#ecmascript-2015-features-destructuring)
meaning this is basically
```
const amount = action.payload === undefined ? 1 : action.payload
```

You can test all these new features here: https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYewdgzgLgBA3gQwLYgK5lgXhgRgL4zYLBQCW4AdAA4ICeANiAgCZA&debug=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=false&presets=es2015%2Creact%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3%2Ces2015-loose%2Cenv&prettier=false&targets=&version=6.26.0